tests/ha/ha_cluster_init.pm

```diff
@@ -34,8 +45,20 @@ sub run {
     # Note: the default timeout need to be increase because it can takes time to join the cluster
     # Initialize the cluster with diskless or shared storage SBD (default)
     $fencing_opt = '-S' if (get_var('USE_DISKLESS_SBD'));
-    if (script_run "ha-cluster-init -y $fencing_opt $unicast_opt", $join_timeout) {
-        assert_script_run "crm -dR cluster init -y $fencing_opt $unicast_opt", $join_timeout;
+    type_string "ha-cluster-init -y $fencing_opt $unicast_opt $qdevice_opt ; echo ha-cluster-init-finished-\$? > /dev/$serialdev\n";
+    if (get_var('QDEVICE') && (check_screen 'ha-cluster-init_qdevice-password', $join_timeout)) {
+        type_password;
+        send_key 'ret';
+    }
+
+    if (!wait_serial("ha-cluster-init-finished-0", $join_timeout)) {
+        type_string "crm -dR cluster init -y $fencing_opt $unicast_opt $qdevice_opt ; echo ha-cluster-init-finished-\$? > /dev/$serialdev\n";
+        if (get_var('QDEVICE')) {
+            assert_screen 'ha-cluster-init_qdevice-password', $join_timeout;
+            type_password;
+            send_key 'ret';
+        }
+        die "Cluster initialization failed" if (!wait_serial("ha-cluster-init-finished-0", $join_timeout));
     }
 
```

Comment:
Question for another PR: in fact I think it maybe better to test both commands (`ha-cluster-init` and `crm init`) on different cluster with a variable?

Reply:
> Question for another PR: in fact I think it maybe better to test both commands (`ha-cluster-init` and `crm init`) on different cluster with a variable?

Don't know. Last time we raised the issue, developers recommended using the bootstrap scripts.

Reply:
> Question for another PR: in fact I think it maybe better to test both commands (`ha-cluster-init` and `crm init`) on different cluster with a variable?

I don't know either as `ha-cluster-init` is just a basic wrapper around `crmsh init "$@"` I'm not it makes sense to test both and duplicate thing.

Reply:
I remember that I had to add this at the beginning because there was an issue with `ha-cluster-init`, but I forgot to add a bsc# entry here.. (I was not aware of this at this time).

So yes, maybe better to remove this `else` part with `crmsh init`. In case of failure better to stop here and open a bug.